Source/API/CBLRevision.m

```diff
@@ -367,12 +367,28 @@ - (void) _addAttachment: (CBLAttachment*)attachment named: (NSString*)name {
     [atts setValue: attachment forKey: name];
     _properties[@"_attachments"] = atts;
     attachment.name = name;
-    attachment.revision = self;
+    
+    // NOTE: Not setting the revision to the attachment object (attachment.revision = self) as
+    // [1] The UnsavedRevision object is not used during save operation
+    // [2] Setting the UnsavedRevision object here will cause the circular reference memory leak.
+    // The revision object will be set in the overridden attachmentName: method.
 }
 
 - (void) removeAttachmentNamed: (NSString*)name {
     [self _addAttachment: nil named: name];
 }
 
+- (CBLAttachment*) attachmentNamed: (NSString*)name {
+    CBLAttachment* base = self.attachmentMetadata[name];
```

Comment:
I looked at the super class but as I didn't set the the revision to the attachment object, the super class will return the same CBLAttachment without revision object set.